python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py

```diff
@@ -120,6 +125,31 @@ def create_dagster_pandas_dataframe_description(description, columns):
     return buildme
 
 
+def create_dagster_pandas_dataframe_metadata(
+    pandas_df: pd.DataFrame,
+) -> Union[TypeCheck, MetadataValue]:
+    """
+    This function takes a pandas DataFrame and returns its metadata as a Dagster TableSchema.
+
+    Args:
+        pandas_df (pandas.DataFrame): A pandas DataFrame for which to create metadata.
+
+    Returns:
+        Union[TypeCheck, MetadataValue]: If the input is not a pandas DataFrame, it returns a TypeCheck object
+        with success=False. Otherwise, it returns a MetadataValue object with the TableSchema for the DataFrame.
+    """
+    if not isinstance(pandas_df, pd.DataFrame):
```

Comment:
Ah, I think you've misunderstood the purpose of Dagster's `TypeCheck`-- this is a framework-specific entity used to encapsulate typechecks for op inputs/outputs, not a generic entity used to represent a typecheck for a python function.

Based on my understanding of the purpose of this function, pretty sure the `TypeCheck` case should just be removed. Use `check.inst(pandas_df, pd.DataFrame)` to verify the input is a dataframe and you can drop the conditional.

Reply:
Thanks for details on this topic. Fixed